src/platforms/esp32/main/spi_driver.c

```diff
@@ -68,91 +134,139 @@ void spi_driver_init(GlobalContext *global)
 
 Context *spi_driver_create_port(GlobalContext *global, term opts)
 {
+    TRACE("spi_driver_create_port\n");
     Context *ctx = context_new(global);
 
-    struct SPIData *spi_data = calloc(1, sizeof(struct SPIData));
+    term bus_config = term_get_map_assoc(ctx, opts, BUS_CONFIG_ATOM);
```

Comment:
are we replacing the proplist with a map, can we discuss pros and cons?

Reply:
Internally it is a map.  The documentation says to use prop lists, though technically a user who knows the code could use a map.  When the user supplies a prop list, it automatically gets translated into a map as part of the config verification to ensure that the input parameters are well-formed.  Yes, it does result in a deep copy, but it is only done once, at initialization time.